Classes/Command/CheckLinksCommandController.php

```diff
@@ -0,0 +1,224 @@
+use Neos\Flow\Cli\CommandController;
+use Neos\Flow\Http\ServerRequestAttributes;
+use Neos\Flow\Mvc\ActionRequestFactory;
+use Neos\Flow\Mvc\ActionResponse;
+use Neos\Flow\Mvc\Controller\Arguments;
+use Neos\Flow\Mvc\Controller\ControllerContext;
+use Neos\Flow\Mvc\Exception\InvalidActionNameException;
+use Neos\Flow\Mvc\Exception\InvalidArgumentNameException;
+use Neos\Flow\Mvc\Exception\InvalidArgumentTypeException;
+use Neos\Flow\Mvc\Exception\InvalidControllerNameException;
+use Neos\Flow\Mvc\Routing\Dto\RouteParameters;
+use Neos\Flow\Mvc\Routing\Exception\MissingActionNameException;
+use Neos\Flow\Mvc\Routing\UriBuilder;
+use Neos\Flow\ObjectManagement\Exception\UnresolvedDependenciesException;
+use Neos\Flow\Persistence\Exception\IllegalObjectTypeException;
+use Neos\Neos\Service\LinkingService;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+
+/**
+ * @Flow\Scope("singleton")
+ * @see https://gist.github.com/hhoechtl/9012d455eab52658bbf4
+ */
+class CheckLinksCommandController extends CommandController
+{
+    /**
+     * @Flow\Inject
+     *
+     * @var ResultItemStorage
+     */
+    protected $resultItemStorage;
+
+    /**
+     * @Flow\Inject
+     * @var NodeDataStorage
+     */
+    protected $nodeDataStorage;
+
+    /**
+     * @Flow\Inject
+     * @var LinkingService
+     */
+    protected $linkingService;
+
+    /**
+     * @Flow\Inject
+     * @var ContextFactoryInterface
+     */
+    protected $contextFactory;
+
+    /**
+     * @Flow\Inject
+     * @var NodeFactory
+     */
+    protected $nodeFactory;
+
+    /**
+     * @Flow\Inject
+     * @var ServerRequestFactoryInterface
+     */
+    protected $serverRequestFactory;
+
+    /**
+     * @Flow\Inject
+     * @var ActionRequestFactory
+     */
+    protected $actionRequestFactory;
+
+    protected array $settings;
+
+    /**
+     * Inject the settings
+     *
+     * @param array $settings
+     * @return void
+     */
+    public function injectSettings(array $settings): void
+    {
+        $this->settings = $settings;
+    }
+
+    /**
+     * Crawl for invalid links
+     *
+     * This command crawls an url for invalid internal and external links
+     *
+     * @param string $url The url to start crawling from
+     * @param int $concurrency Maximum number of requests to send concurrently
+     * @return void
+     * @throws InvalidActionNameException
+     * @throws InvalidArgumentNameException
+     * @throws InvalidArgumentTypeException
+     * @throws InvalidControllerNameException
+     * @throws MissingActionNameException
+     * @throws NodeConfigurationException
+     * @throws UnresolvedDependenciesException
+     * @throws IllegalObjectTypeException
+     * @throws \Neos\Flow\Property\Exception
+     * @throws \Neos\Flow\Security\Exception
+     * @throws \Neos\Neos\Exception
+     * @see \Neos\Neos\Fusion\ConvertUrisImplementation::evaluate
+     */
+    public function crawlCommand(string $url = '', int $concurrency = 10): void
```

Comment:
@rolandschuetz This command looks still quite messy, I will clean it up once I get the first feedback if it goes into the right direction but for now, it finds broken nodes and adds them to the database.